src/bindings/samp.h

```diff
@@ -6142,6 +6142,37 @@ static PyObject* pysamp_getvehiclemodelinfo(PyObject *self, PyObject *args)
 	return out;
 }
 
+static PyObject* logprintf_write(PyObject *self, PyObject *args)
+{
+	const char *text;
+
+	if(!PyArg_ParseTuple(args, "es", "cp1252", &text))
+		return NULL;
+
+	if(
+		strlen(text) == 0
+		|| (
+			strlen(text) == 1
+			&& text[0] == '\n'
+		)
+	)
+	{
+		// We will not print an extra \n in the logs
+		PyMem_Free((void *)text);
+		return PyLong_FromUnsignedLong(0);
+	}
+
+	unsigned int written_count = strlen(text);
+	sampgdk::logprintf(text);
+	PyMem_Free((void *)text);

```

Comment:
Is this really needed?
I just looked at the docs:
"Frees the memory block pointed to by p, which must have been returned by a previous call to PyMem_Malloc(), PyMem_Realloc() or PyMem_Calloc(). Otherwise, or if PyMem_Free(p) has been called before, undefined behavior occurs." Seems incorrect here

Reply:
Well - if you look at https://docs.python.org/3/c-api/arg.html#strings-and-buffers in "se" (the format we're using) you will see:

```
PyArg_ParseTuple() will allocate a buffer of the needed size, copy the encoded data into this buffer and adjust *buffer to reference the newly allocated storage. The caller is responsible for calling PyMem_Free() to free the allocated buffer after use.
```

Also trust me from experience, if it was unnecessary it would in fact blow up :rofl:

Reply:
Ah perfect, didn't know that :) That's why I was asking. This means probably that I'll have to change some other code too :D never freed the memory using PyMem_Free ... Which seems really bad as I skim through the code :D